### PR TITLE
Added Header model and to xml conversion

### PR DESCRIPTION
**Pull request by chris-ianson**
Added
- Header
- to xml conversion conversion added for models used in unloading remarks

### CODE REVIEW

File: app/models/TraderAtDestination.scala [hunk 1]
```diff
@@ -55,6 +58,36 @@ object TraderAtDestinationWithEori {
     val cityLength            = 35
     val countryCodeLength     = 2
   }
+
+  implicit def writes: XMLWrites[TraderAtDestinationWithEori] = XMLWrites[TraderAtDestinationWithEori] {
```
Comment (garethgooch) [inline app/models/TraderAtDestination.scala:62]: Are able to have this, don't we require this on the arrival so should (unless core will strip this) have this for every unloading permission?

File: test/generators/Generators.scala [hunk 2]
```diff
@@ -146,10 +146,6 @@ trait Generators extends UserAnswersGenerator with PageGenerators with ModelGene
     }
   }
 
-  implicit lazy val arbitraryLocalDate: Arbitrary[LocalDate] = Arbitrary {
```
Comment (garethgooch) [inline test/generators/Generators.scala:149]: ??

File: test/generators/MessagesModelGenerators.scala [hunk 3]
```diff
@@ -68,4 +70,17 @@ trait MessagesModelGenerators extends Generators {
...
     }
   }
+
+  implicit lazy val arbitraryHeader: Arbitrary[Header] = {
+    Arbitrary {
+      for {
+        movementReferenceNumber <- stringsWithMaxLength(UnloadingPermission.movementReferenceNumberLength)
+        transportIdentity       <- Gen.option(stringsWithMaxLength(UnloadingPermission.transportIdentityLength))
+        transportCountry        <- Gen.option(stringsWithMaxLength(UnloadingPermission.transportCountryLength))
+        numberOfItems           <- choose(min = 1: Int, 2: Int)
```
Comment (garethgooch) [inline test/generators/MessagesModelGenerators.scala:80]: are we happy with this limit of upto 2 items?

File: test/models/ProducedDocumentSpec.scala [hunk 4]
```diff
@@ -48,18 +49,51 @@ class ProducedDocumentSpec extends FreeSpec with MustMatchers with Generators wi
           }
 
           val expectedResult = {
-            <SGICODSD2>
+            <PRODOCDC2>
```
Comment (garethgooch) [inline test/models/ProducedDocumentSpec.scala:52]: this seems like a big change

File: test/models/TraderAtDestinationSpec.scala [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+            city => <CitTRD24>{city}</CitTRD24>
+          )
+          val countryCodeNode = trader.countryCode.map(
+            countryCode => <CouTRD25>{countryCode}</CouTRD25>
+          )
+
+          val expectedResult =
+            <TRADESTRD>
+              {
+                nameNode.getOrElse(NodeSeq.Empty) ++
```
Comment (garethgooch) [inline test/models/TraderAtDestinationSpec.scala:54]: not a fan of get or else here, are we able to refactor

File: app/models/Seals.scala [hunk 6]
```diff
@@ -32,4 +35,20 @@ object Seals {
     (__ \ "SEAIDSID" \ "SeaIdeSID1").read(seq[String])
   ).mapN(apply)
 
+  implicit def writes: XMLWrites[Seals] = XMLWrites[Seals] {
```
[the remark this answers is not in the document]
Comment (chris-ianson) [reply app/models/Seals.scala:38]: I know what you mean but by excluding _SEAINFSLI_ from the reader tree is a way of telling  the reader it's optional. If no seals exist don't throw a parse error (which is what happens if you put it in and no seals are returned within the XML) The writer is slightly different and will only ever be called when we know there's seals

File: app/models/TraderAtDestination.scala (hunk 1 above)
Comment (chris-ianson) [reply app/models/TraderAtDestination.scala:62]: It is on the arrival and comes through on unloading permission. We don't do anything with the data, just pass it on in unloading remarks

File: test/generators/Generators.scala (hunk 2 above)
Comment (chris-ianson) [reply test/generators/Generators.scala]: Yeah, I removed it. Rebased master and it's there now.

File: test/generators/MessagesModelGenerators.scala (hunk 3 above)
Comment (chris-ianson) [reply test/generators/MessagesModelGenerators.scala:80]: I'm not sure how far to go with this. I mean the spec says you can have 9999x?! What do you think?

File: test/models/ProducedDocumentSpec.scala (hunk 4 above)
Comment (chris-ianson) [reply test/models/ProducedDocumentSpec.scala:52]: Incorrect node

File: test/models/TraderAtDestinationSpec.scala (hunk 5 above)
Comment (chris-ianson) [reply test/models/TraderAtDestinationSpec.scala:54]: How come? We could to a fold on where we set it and default to NodeSeq.Empty (some of them are like this)

**Pull request merged**